docs/server_rest_api.md

```diff
@@ -0,0 +1,675 @@
+---
+title: Guardrails API v0.0.0
+language_tabs:
+  - python: Python
+toc_footers: []
+includes: []
+search: true
+highlight_theme: darkula
+headingLevel: 2
+
+---
+<!-- this was generated with widdershins --environment env.json service-specs/guardrails-service-spec.yml -o api_ref01.md --expandBody true -->
+<!-- Generator: Widdershins v4.0.1 -->
+
+<h1 id="guardrails-api">Guardrails Server API</h1>
+
+> A guardrails API server can be iniated via the CLI with `guardrails start`. It can be interacted with via `Guard(`.  This is reference desribes the underlying http interface the client interacts with the server via.
+
+Guardrails CRUD API
+
+# Authentication
+
+- HTTP Authentication, scheme: bearer 
+
+* API Key (ApiKeyAuth)
+    - Parameter Name: **authorization**, in: header. 
+
+<h1 id="guardrails-api-service-health">Service Health</h1>
+
+## get__health-check
+
+`GET /health-check`
+
+> Example responses
+
+> 200 Response
+
+```json
+{
+  "status": 0,
+  "message": "string"
+}
+```
+
+<h3 id="get__health-check-responses">Responses</h3>
+
+|Status|Meaning|Description|Schema|
+|---|---|---|---|
+|200|[OK](https://tools.ietf.org/html/rfc7231#section-6.3.1)|Returns the status of the server|[HealthCheck](#schemahealthcheck)|
+|default|Default|Unexpected error|[HttpError](#schemahttperror)|
+
+<aside class="warning">
+To perform this operation, you must be authenticated by means of one of the following methods:
+ApiKeyAuth, BearerAuth
+</aside>
+
+<h1 id="guardrails-api-guard">guard</h1>
+
+## getGuards
+
+<a id="opIdgetGuards"></a>
+
+`GET /guards`
+
+*Fetches the configuration for all Guards the user has access to.*
+
+> Example responses
+
+> 200 Response
+
+```json
+[]
+```
+
+<h3 id="getguards-responses">Responses</h3>
+
+|Status|Meaning|Description|Schema|
+|---|---|---|---|
+|200|[OK](https://tools.ietf.org/html/rfc7231#section-6.3.1)|A list of Guards.|Inline|
```

Comment:
This isn't showing any of the actual schemas.  Maybe try re-generating this with the fully hydrated OpenAPI Spec that's published as part of the guardrails_api module.

Reply:
you mean guardrails_api/open-api-spec.json?

Reply:
does that come from guardrails-api-client/service-specs/guardrails-service-spec.yml ?

Reply:
Kinda of, it's built from that but this one is fully hydrated.  You're right that the proper source of truth would come from the `guardrails_api_client` module.  Relative path would be `guardrails_api_client/openapi-spec.json`